crates/shared/src/testing/consensus.rs

```diff
@@ -0,0 +1,170 @@
+//! This module defines types used when simulating consensus in tests
```

Comment:
Extracts a bunch of logic related to simulating consensus behaviour we've been using in tests. Basically everything not directly related to the builder.